### PR TITLE
Add action to check links

### LINKED ISSUE #22: add linter for dead links
I have a habit of linking to things that don't exist yet, which can lead me to merging dead links if I'm not careful.

There are a few tools we could use to lint the source or generated output and check for dead links.

https://github.com/wjdp/htmltest seems promising.

We should add it to circle and fail if internal links are broken.

I can't decide whether broken external links should fail CI - it feels flaky to have the build fail for something outside our control. Maybe it could be a warning? It would be nice to be aware when external links disappear.

### PR DESCRIPTION
**Pull request by DannyS03**
## Context

- Quick config for broken and dead links
> uses [this link check](https://github.com/lycheeverse/lychee-action)
- resolves #22

## Latest preview

Please view CI

### CODE REVIEW

File: content/concepts/protocols.md [hunk 1]
```diff
@@ -105,16 +105,8 @@ all of the registered match functions. If any returns `true`, the associated han
 This gives you a lot of flexibility to do your own "fuzzy matching" and define whatever rules for protocol matching
 make sense for your application.
 
-#### Match using semver
```
Comment (marten-seemann) [inline content/concepts/protocols.md:108]: Can this be a separate PR?

File: .github/workflows/link-check.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  workflow_dispatch:
+
+jobs:
+  linkChecker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Link Checker
+        uses: lycheeverse/lychee-action@v1
```
Comment (marten-seemann) [inline .github/workflows/link-check.yml:13]: To avoid supply-chain attacks, we should include by commit hash and not by version.
```suggestion
        uses: lycheeverse/lychee-action@4a5af7cd2958a2282cefbd9c10f63bdb89982d76 # v1.5.1
```

File: content/concepts/protocols.md (hunk 1 above)
Comment (DannyS03) [reply content/concepts/protocols.md]: Yes

**Comment by DannyS03** [comment]
Thanks @marten-seemann! Separate PR for remaining link: https://github.com/libp2p/docs/pull/185

**Review by marten-seemann — APPROVED** [review]
